functions/Restore-DbaSystemDatabase.ps1

```diff
@@ -0,0 +1,367 @@
+        }
+        catch {
+            Stop-Function -message "Cannot connect to $SqlInstance, stopping" -target $SqlInstance 
+            return
+        }
+        if (($server.IsCluster) -and $AcceptClusterRisk -ne $true) {
+            Stop-Function -message "Clustered Instance and AcceptClusterRisk not set, `n please read notes in help and decide how to proceed"
+            return
+        }
+        $CurrentStartup = Get-DbaStartupParameter -SqlInstance $SqlInstance
+        if ((Get-DbaSqlService -ComputerName $ServerName -Credential $Credential -Type Agent | Where-Object {$_.DisplayName -like "*$InstanceName*"}).State -eq 'Running') {
+            Write-Message -Level Verbose -Message "SQL agent running, stopping it" 
+            $RestartAgent = $True
+            Stop-DbaService -SqlInstance $server -service SqlAgent | out-null
+        }
+        if ((Get-DbaSqlService -ComputerName $ServerName -Credential $Credential -Type FullText | Where-Object {$_.DisplayName -like "*$InstanceName*"}).State -eq 'Running') {
+            Write-Message -Level Verbose -Message "Full Text agent running, stopping it" 
+            $RestartFullText = $True
+            Stop-DbaService -SqlInstance $server -service FullText | out-null
+        }
+        try {
+            if ($true -eq $master) {
+            
+                Write-Message -Level Verbose -Message "Restoring Master, setting single user"
+                Set-DbaStartupParameter -SqlInstance $server -SingleUser -SingleUserDetails dbatoolsSystemk34i23hs3u57w | out-null
+                Stop-DbaService -SqlInstance $server | out-null
+                Start-DbaService -SqlInstance $server | out-null
+                $StartCount = 0
+                while ((Get-DbaSqlService -ComputerName $ServerName -Credential $Credential -Type Engine | Where-Object {$_.DisplayName -like "*$InstanceName*"}).State -ne 'running') {
+                    Start-DbaService -SqlInstance $server | out-null
+                    Start-Sleep -seconds 65
+                    $StartCount++
+                    if ($StartCount -ge 4) {
+                        #Didn't start nicely, jump to finally to try to come back up sanely
+                        Stop-Function -Message "SQL Server not starting nicely, tried to fix, but not responding" 
+                        return
+                    }
+                }
+                if ($server.connectionContext.IsOpen -eq $false) {
+                    $server.connectionContext.Connect()
+                }
+                Write-Message -Level Verbose  -Message  "Beginning Restore of Master"
+                if ($pscmdlet.ShouldProcess("Restoring Master on $ServerName")) {
+                    $RestoreResult += Restore-DbaDatabase -SqlInstance $server -Path $BackupPath -WithReplace -DatabaseFilter master -RestoreTime $RestoreTime -ReuseSourceFolderStructure -SystemRestore           
```

Comment:
Should failure here interrupt execution?
If yes, you may want to add the `-Silent` parameter, so it throws and the catch block triggers